hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -69,11 +69,17 @@ func TunedConfigMap(namespace, name string) *corev1.ConfigMap {
 	}
 }
 
-func PerformanceProfileConfigMap(namespace, name string) *corev1.ConfigMap {
+func PerformanceProfileConfigMap(namespace, name, nodePoolName string) *corev1.ConfigMap {
+	fullName := fmt.Sprintf("%s-%s", name, nodePoolName)
+	// 63 is the qualifiedNameMaxLength allowed for k8s object
+	// https://github.com/kubernetes/kubernetes/blob/957c9538670b5f7ead2c9ba9ceb9de081d66caa4/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L34
+	if len(fullName) > 63 {
+		fullName = fmt.Sprintf("%s-%s", name[:32], nodePoolName[:31])
```

Comment:
You have not checked the length of `name` or `nodePoolName` so these slices may be out-of-bounds and panic.

Also, I would like to gently push back on your comment [here](https://github.com/openshift/hypershift/pull/4108/files#r1623367155) about "over complication" - if we do not write this code correctly, the controller will fail to function and human intervention will be required. I don't see a compelling reason to choose to write a method here that fails to correctly encode the name and hope that we don't hit those cases in production.

Reply:
> You have not checked the length of name or nodePoolName so these slices may be out-of-bounds and panic.

My bad. Thank you for that

> Also, I would like to gently push back on your comment [here](https://github.com/openshift/hypershift/pull/4108/files#r1623367155) about "over complication" - if we do not write this code correctly, the controller will fail to function and human intervention will be required. I don't see a compelling reason to choose to write a method here that fails to correctly encode the name and hope that we don't hit those cases in production.

I apologies if my comment sound a bit rude, that's not what I meant.
What I meant is that having a weighted prefixes, where each name gets its prefix in proportion to its total length, is a bit complex because there are many edge cases and in some of them, the name still might be ambiguous.

BTW the previous implementation was problematic as well, because we are not checking the length of `perfprofile-+name`

Reply:
I see, no offense taken. Non-weighted approaches do exist, as well. I don't have any preference and the weighted approach is very reasonable.